File: src/agente_deposito.py
from src.usuario import Usuario

class AgenteDeposito(Usuario):

    def __init__(self, id_usuario: int, nombre: str,
                 email: str, contrasena: str, deposito: str):
        super().__init__(id_usuario, nombre, email, contrasena)
        self.__deposito = deposito

    def login(self) -> bool:
        print(f"{self.get_nombre()} ha iniciado sesión")
        return True

    def logout(self) -> None:
        print(f"{self.get_nombre()} ha cerrado sesión")

    def consultar_ordenes(self) -> list:
        print(f"{self.get_nombre()} está consultando órdenes")
        return []

    def crear_pedido(self):
        print(f"{self.get_nombre()} está creando un pedido")

    # --- Getter ---
    def get_deposito(self) -> str:
        return self.__deposito

    # --- Setter ---
    def set_deposito(self, deposito: str):
        self.__deposito = deposito

File: Televentas Java/Clases/AgenteDeposito.java
package Clases;

import java.util.ArrayList;
import java.util.List;

public class AgenteDeposito extends Usuario {

    private String deposito;

    public AgenteDeposito(int id, String nombre, String email,
                          String contrasena, String deposito) {
        super(id, nombre, email, contrasena);
        this.deposito = deposito;
    }

    @Override
    public boolean login() {
        System.out.println(getNombre() + " ha iniciado sesión");
        return true;
    }

    @Override
    public void logout() {
        System.out.println(getNombre() + " ha cerrado sesión");
    }

    public List<Object> consultarOrdenes() {
        System.out.println(getNombre() + " está consultando órdenes");
        return new ArrayList<>();
    }

    public void crearPedido() {
        System.out.println(getNombre() + " está creando un pedido");
    }

    // --- Getter ---
    public String getDeposito() { return deposito; }

    // --- Setter ---
    public void setDeposito(String deposito) { this.deposito = deposito; }
}
